hypervisor/network/network_linux.go

```diff
@@ -1013,11 +1014,115 @@ func Allocate(vmId, requestedIP string, index int, addrOnly bool, maps []pod.Use
 		IPPrefixLen: maskSize,
 		Device:      device,
 		File:        tapFile,
+		Automatic:   true,
+	}, nil
+}
+
+func Configure(vmId, requestedIP string, addrOnly bool,
+	maps []pod.UserContainerPort, config pod.UserInterface) (*Settings, error) {
+	var (
+		req   ifReq
+		errno syscall.Errno
+	)
+
+	ip, ipnet, err := net.ParseCIDR(config.Ip)
+	if err != nil {
+		glog.Errorf("Parse config IP failed %s", err)
+		return nil, err
+	}
+
+	BridgeIface := config.Bridge
+	maskSize, _ := ipnet.Mask.Size()
+
+	err = SetupPortMaps(ip.String(), maps)
+	if err != nil {
+		glog.Errorf("Setup Port Map failed %s", err)
+		return nil, err
+	}
+
+	mac := config.Mac
+	if mac == "" {
+		mac, err = GenRandomMac()
+		if err != nil {
+			glog.Errorf("Generate Random Mac address failed")
+			return nil, err
+		}
+	}
+
+	if addrOnly {
+		return &Settings{
+			Mac:         mac,
+			IPAddress:   ip.String(),
+			Gateway:     config.Gw,
+			Bridge:      config.Bridge,
+			IPPrefixLen: maskSize,
+			Device:      "",
+			File:        nil,
+			Automatic:   false,
+		}, nil
+	}
+
+	tapFile := new(os.File)
+	tapFile, err = os.OpenFile("/dev/net/tun", os.O_RDWR, 0)
```

Comment:
the `tapFile := new(os.File)` is unneeded
`tapFile, err := os.OpenFile("/dev/net/tun", os.O_RDWR, 0)` is enough. (note the `:=`)